### PR TITLE
Rpc/declare v2

### PR DESCRIPTION
**Pull request by marioiordanov**
## Usage related changes

RPC declare V2

## Checklist:

- [x] Applied formatting - `./scripts/format.sh`
- [x] No linter errors - `./scripts/clippy_check.sh`
- [ ] Performed code self-review
- [x] Rebased to the last commit of the target branch (or merged it into my branch)
- [x] Documented the changes
- [x] Linked the issues which this PR resolves
- [x] Updated the tests
- [x] All tests are passing - `cargo test`

### CODE REVIEW

File: crates/starknet/src/transactions/declare_transaction_v2.rs [hunk 1]
```diff
@@ -20,6 +20,30 @@ pub struct DeclareTransactionV2 {
...
+            signature,
+            nonce,
+            class_hash: None,
+            transaction_hash: None,
+            chain_id,
+        }
+    }
+}
+
 impl DeclareTransactionV2 {
```
Comment (FabijanC) [inline crates/starknet/src/transactions/declare_transaction_v2.rs:47]: Should the two `impl` blocks be unified?

File: crates/starknet-server/src/api/json_rpc/write_endpoints.rs [hunk 2]
```diff
@@ -124,6 +129,20 @@ fn convert_to_deploy_account_transaction(
...
+    DeclareTransactionV2::new(
+        ContractClass::from(value.contract_class),
+        value.compiled_class_hash.0,
+        value.sender_address.0,
+        value.common.max_fee.0,
+        value.common.signature.iter().map(|x| x.0).collect(),
+        value.common.nonce.0,
+        chain_id,
+    )
+}
```
Comment (ivpavici) [inline crates/starknet-server/src/api/json_rpc/write_endpoints.rs:145]: new line here?

File: crates/starknet/src/transactions/declare_transaction_v2.rs (hunk 1 above)
Comment (marioiordanov) [reply crates/starknet/src/transactions/declare_transaction_v2.rs]: changed in [71f0158](https://github.com/0xSpaceShard/starknet-devnet-rs/pull/78/commits/71f01582630b8520e595a679da97e79c1e183fd3)

File: crates/starknet-server/src/api/json_rpc/write_endpoints.rs (hunk 2 above)
Comment (marioiordanov) [reply crates/starknet-server/src/api/json_rpc/write_endpoints.rs:147]: fixed

**Pull request merged**